presto-benchmark-runner/src/test/java/com/facebook/presto/benchmark/source/TestDbBenchmarkSuiteSupplier.java

```diff
@@ -0,0 +1,80 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.benchmark.source;
+
+import com.facebook.presto.testing.mysql.TestingMySqlServer;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.facebook.airlift.testing.Closeables.closeQuietly;
+import static com.facebook.presto.benchmark.BenchmarkTestUtil.BENCHMARK_QUERIES_TABLE;
+import static com.facebook.presto.benchmark.BenchmarkTestUtil.BENCHMARK_SUITES_TABLE;
+import static com.facebook.presto.benchmark.BenchmarkTestUtil.getBenchmarkSuiteObject;
+import static com.facebook.presto.benchmark.BenchmarkTestUtil.getJdbi;
+import static com.facebook.presto.benchmark.BenchmarkTestUtil.insertBenchmarkQuery;
+import static com.facebook.presto.benchmark.BenchmarkTestUtil.insertBenchmarkSuite;
+import static com.facebook.presto.benchmark.BenchmarkTestUtil.setupMySql;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestDbBenchmarkSuiteSupplier
+{
+    private static final String SUITE = "test_suite";
+    private static final String QUERY_SET = "test_set";
+    private static TestingMySqlServer mySqlServer;
+    private static Handle handle;
+    private static BenchmarkSuiteConfig config;
+    private static Jdbi jdbi;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        mySqlServer = setupMySql();
+        jdbi = getJdbi(mySqlServer);
+        handle = jdbi.open();
+        config = new BenchmarkSuiteConfig()
+                .setSuitesTableName(BENCHMARK_SUITES_TABLE)
+                .setQueriesTableName(BENCHMARK_QUERIES_TABLE);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy()
+    {
+        closeQuietly(mySqlServer, handle);
+    }
+
+    @AfterMethod
+    public void cleanup()
+    {
+        handle.execute("DELETE FROM benchmark_suites");
+        handle.execute("DELETE FROM benchmark_queries");
+    }
+
+    @Test
+    public void testSupplySuite()
+    {
+        insertBenchmarkQuery(handle, QUERY_SET, "Q1", "SELECT 1");
+        insertBenchmarkQuery(handle, QUERY_SET, "Q2", "SELECT 2");
+        insertBenchmarkQuery(handle, QUERY_SET, "Q3", "SELECT 3");
+
+        insertBenchmarkSuite(handle, SUITE, QUERY_SET, "[{\"executionStrategy\": \"STREAM\",\"streamConfiguration\": [[\"Q1\", \"Q2\"],[\"Q2\", \"Q3\"]]}, {\"executionStrategy\": \"CONCURRENT\",\"streamConfiguration\": [[\"Q2\", \"Q3\"],[\"Q1\",\"Q2\"]]}]", "{\"max\":\"5\"}");
```

Comment:
What's the difference between STREAM and CONCURRENT here? I thought CONCURRENT means the queries will be executed at the same time. Does it mean Q2 and Q3 will be executed concurrently and Q1 and Q2 will be executed concurrently later?

Reply:
My mistake, I coded it incorrectly. You are right, CONCURRENT means all the queries will be executed at the same time. 
CONCURRENT - ["Q1", "Q2", "Q3"]
STREAMS - [["Q1","Q2"],["Q2","Q3"]]
I made changes accordingly.